pkg/webui/console/store/reducers/user-preferences.js

```diff
@@ -12,20 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { GET_BOOKMARKS_LIST_SUCCESS } from '@console/store/actions/user-preferences'
+import {
+  ADD_BOOKMARK_SUCCESS,
+  GET_ALL_BOOKMARKS_SUCCESS,
+  GET_BOOKMARKS_LIST_SUCCESS,
+} from '@console/store/actions/user-preferences'
 import { GET_USER_ME_SUCCESS } from '@console/store/actions/logout'
 
+// Update a range of values in an array by using another array and a start index.
+const fillIntoArray = (array, start, values, totalCount) => {
```

Comment:
Please don't copy/paste functions but promote them to a utility module and then use it throughout.